exercicios/para-casa/sistema_de_alunas.py

```diff
@@ -1,59 +1,140 @@
+            cod_opcao = int(input("\nEscolha uma opção:\n"
+                                  "1 - Incluir uma nova aluna\n"
+                                  "2 - Consultar lista de alunas\n"
+                                  "3 - Consultar faltas da aluna\n"
+                                  "4 - Consultar notas da aluna\n"
+                                  "5 - Consultar status de aprovação\n"
+                                  "6 - Sair do sistema\n"
+                                  "Opção: "))
+            if cod_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+            else:
+                return cod_opcao
         except ValueError:
             print("Entrada inválida. Por favor, digite um número inteiro.\n")
-            
-        return codigo_opcao
-    
-def incluir_nova_aluna():
-    pass
-    #TODO - Implentar a função
+
+
+def obter_dados():
+    nome = input('Qual o nome da nova aluna?: ').title()
+    sobrenome = input(f'Qual sobrenome da {nome}?: ').title()
+    turma = input(f'Qual turma a {nome, sobrenome} pertence?: ')
+    lista_presença = obter_presença()
+    nota_participação = float(input('Participação da nova aluna: '))
+    notas = obter_notas()
+
+    salvar_dados(nome, sobrenome, turma, lista_presença, nota_participação, notas)
+
+def obter_notas():
+    quantidade_notas = int(input('Quantas notas serão calculadas?: '))
+    notas = []
+    for contador in range(quantidade_notas):
+        while True: 
+            entrada = input(f'Nota do #{contador + 1}: ')
+            try:
+                nota = float(entrada)
+                notas.append(nota)
+                break
+            except ValueError:
+                print('Erro em processar o valor. Insira um número válido')
     
+    return notas
+
+def obter_presença():
+    quantidade_aulas = int(input('Digite quantas aulas houveram: '))
+    lista_presença = []
+
+    for contador in range(quantidade_aulas):
+        while True:
+            entrada = input(f'Presença na aula #{contador +1} (True/False): ')
+            if entrada in ['True', 'False']:
+                lista_presença.append(eval(entrada))
+                break
+            else:
+                print('Informação inválida. Insira apenas True ou False')
+
+    return lista_presença
+
+def salvar_dados(nome, sobrenome, turma, lista_presença, nota_participação, notas):
+    chave = (nome, sobrenome)
+    dataset[chave] = { 
+        "Turma": turma,
+        "Notas": notas,
+        "Presença": lista_presença,
+        "Participação": nota_participação
+    }
+    print(f'Os dados da aluna {nome} {sobrenome} foram inseridos com sucesso.')
+
 def consultar_lista_alunas():
-    pass
-    #TODO - Implentar a função
-    
+    if dataset:
+        print('Lista de todas as alunas')
+        for chave in dataset.keys():
+            print(f'Nome: {chave[0]} {chave[1]}')
```

Comment:
Uso da tupla conforme o esperado. Uma opção para trazer maior clareza ao código é desempacotar a tupla em variáveis que fiquem mais fáceis.

**Sugestão:**
```suggestion
    for nome, sobrenome in dataset.keys():
        print(f'Nome: {nome} {sobrenome}')
```